digiwf-oce-cosmos-integration-service/src/main/java/io/muenchendigital/digiwf/ocecosmos/integration/api/streaming/OceCosmosStreamingEventListener.java

```diff
@@ -67,12 +71,73 @@ public Consumer<Message<JobRequestDto>> requestPrint() {
         };
     }
 
-    private Path convertByteArrayToFile(JobRequestDto printRequest, FileSystem fileSystem) throws IOException {
+    /**
+     * The Consumer expects an {@link FileJobRequestDto}.
+     * <p>
+     * After successfully requesting the printing job a JSON representing a {@link JobResponse} is returned.
+     * <p>
+     * In case of an error the error message is returned as a JSON representing {@link OceCosmosErrorDto}.
+     */
+    @Bean
+    public Consumer<Message<FileJobRequestDto>> requestPDFPrint() {
+        return message -> {
+            log.debug(message.toString());
+
+            final var printRequest = message.getPayload();
+
+            Object printJobResult;
+
+            if(printRequest.getS3Files() == null) {
+                this.correlateMessageService.sendCorrelateMessage(
+                        message.getHeaders(),
+                        Map.of(RESPONSE, new OceCosmosErrorDto("Presigned S3 File-URLs needed"))
+                );
+                return;
+            }
+
+            if(printRequest.getS3Files().size() != 1) {
```

Comment:
Why use a list if only one file is supported?

Reply:
The s3 integration always returns an array, so to reduce the js mapping code in the workflow, i accept an array here. Beside that multi file support may be needed in the future